jsm.go

```diff
@@ -679,3 +778,90 @@ func (s *StreamLister) Err() error {
 func (js *js) NewStreamLister() *StreamLister {
 	return &StreamLister{js: js}
 }
+
+type streamNamesLister struct {
+	js *js
+
+	err      error
+	offset   int
+	page     []string
+	pageInfo *apiPaged
+}
+
+// streamNamesListResponse is the response for a Streams Names List request.
+type streamNamesListResponse struct {
+	apiResponse
+	apiPaged
+	Streams []string `json:"streams"`
+}
+
+// Next fetches the next ConsumerInfo page.
+func (l *streamNamesLister) Next() bool {
+	if l.err != nil {
+		return false
+	}
+	if l.pageInfo != nil && l.offset >= l.pageInfo.Total {
+		return false
+	}
+
+	subj := l.js.apiSubj(fmt.Sprintf(apiStreamNamesT))
+	r, err := l.js.nc.Request(subj, nil, l.js.wait)
```

Comment:
Seems this is the same request as this one but with an empty payload? https://github.com/nats-io/nats.go/blob/master/js.go#L655-L676

Reply:
Good catch. I deleted `apiStreamNamesT` and `streamNamesListResponse`.